pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -44,63 +46,137 @@ func NewSecretGenerator(client client.Client, apiReader client.Reader, dtClient
 func (s *SecretGenerator) GenerateForDynakube(ctx context.Context, dk *dynakube.DynaKube) error {
 	log.Info("reconciling namespace bootstrapper init secret for", "dynakube", dk.Name)
 
-	data, err := s.generate(ctx, dk)
+	// TODO: only metadata-enrichment => only dynatrace-bootstrapper-config,
+	//  and only the endpoint.propertiespart, rest is not needed
+	data, err := s.generateConfig(ctx, dk)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	err = s.createSourceForWebhook(ctx, dk, data)
+	err = s.createSourceForWebhook(ctx, dk, GetSourceConfigSecretName(dk.Name), ConfigConditionType, data)
 	if err != nil {
 		return err
 	}
 
 	nsList, err := mapper.GetNamespacesForDynakube(ctx, s.apiReader, dk.Name)
 	if err != nil {
-		conditions.SetKubeApiError(dk.Conditions(), ConditionType, err)
+		conditions.SetKubeApiError(dk.Conditions(), ConfigConditionType, err)
 
 		return errors.WithStack(err)
 	}
 
+	err = s.createSecretForNSlist(ctx, consts.BootstrapperInitSecretName, ConfigConditionType, nsList, dk, data)
+
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	certs, err := s.generateCerts(ctx, dk)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// If there are no certs, we don't need to create the certs secret
+	// TODO: not clear for now should we replicate old logic with empty data inside secrets object
```

Comment:
will TODO will be done separately?

Reply:
clear up: empty data should not be sent

Reply:
removed TODO, since already implemented in  ffbd796ce